### PR TITLE
Fully map out block state model generator

### PR DESCRIPTION
**Pull request by liach**
Now eliminated all unmapped classes! :tada:

Signed-off-by: liach <liach@users.noreply.github.com>

### CODE REVIEW

File: mappings/net/minecraft/data/client/model/BlockStateModelGenerator.mapping [hunk 1]
```diff
@@ -1,12 +1,422 @@
...
+	METHOD method_25508 registerChorusPlant ()V
+	METHOD method_25509 registerComposter ()V
+	METHOD method_25510 registerDaylightDetector ()V
+	METHOD method_25511 registerEndRod ()V
+	METHOD method_25512 registerFarmland ()V
+	METHOD method_25513 registerFire ()V
+	METHOD method_25514 registerSoulFire ()V
+	METHOD method_25515 registerFrostedIce ()V
+	METHOD method_25516 registerTopSoils ()V
+	METHOD method_25517 registerCocoaBeans ()V
```
Comment (YanisBft) [inline mappings/net/minecraft/data/client/model/BlockStateModelGenerator.mapping]: CocoaBeans is the item, the block is just Cocoa

File: mappings/net/minecraft/data/client/model/Texture.mapping [hunk 2]
```diff
@@ -1,20 +1,20 @@
 CLASS net/minecraft/class_4944 net/minecraft/data/client/model/Texture
 	FIELD field_22997 entries Ljava/util/Map;
 	FIELD field_22998 inherited Ljava/util/Set;
-	METHOD method_25860 getModelId (Lnet/minecraft/class_2248;)Lnet/minecraft/class_2960;
+	METHOD method_25860 getId (Lnet/minecraft/class_2248;)Lnet/minecraft/class_2960;
```
Comment (YanisBft) [inline mappings/net/minecraft/data/client/model/Texture.mapping:4]: Why this change? Specifying that it's the model id make it more clear

Comment (liach) [reply mappings/net/minecraft/data/client/model/Texture.mapping:4]: this can actually be texture id as well

File: mappings/net/minecraft/data/client/model/BlockStateModelGenerator.mapping (hunk 1 above)
Comment (liach) [reply mappings/net/minecraft/data/client/model/BlockStateModelGenerator.mapping]: Done.

**Review by YanisBft — APPROVED** [review]

File: mappings/net/minecraft/data/client/model/BlockStateModelGenerator.mapping [hunk 3]
```diff
@@ -1,12 +1,422 @@
...
+		ARG 2 ageProperty
+		ARG 3 ageTextureIndices
+	METHOD method_25548 registerTintableCross (Lnet/minecraft/class_2248;Lnet/minecraft/class_4910$class_4913;)V
+		ARG 1 block
+		ARG 2 tintType
+	METHOD method_25549 registerTintableCross (Lnet/minecraft/class_2248;Lnet/minecraft/class_4910$class_4913;Lnet/minecraft/class_4944;)V
+		ARG 1 block
+		ARG 2 tintType
+		ARG 3 texture
+	METHOD method_25550 registerNorthDefaultHorizontalRotatable (Lnet/minecraft/class_2248;Lnet/minecraft/class_4944;)V
```
Comment (Runemoro) [inline mappings/net/minecraft/data/client/model/BlockStateModelGenerator.mapping]: I think `registerHorizontalRotatable`, and adding the fact that north is the default in a javadoc, is better.

File: mappings/net/minecraft/data/client/model/BlockStateModelGenerator.mapping [hunk 4]
```diff
@@ -1,12 +1,422 @@
...
+		ARG 3 snowyVariant
+	METHOD method_25561 registerPiston (Lnet/minecraft/class_2248;Lnet/minecraft/class_2960;Lnet/minecraft/class_4944;)V
+		ARG 1 piston
+		ARG 2 extendedModelId
+		ARG 3 texture
+	METHOD method_25565 createBooleanPropertyBasedModelMap (Lnet/minecraft/class_2746;Lnet/minecraft/class_2960;Lnet/minecraft/class_2960;)Lnet/minecraft/class_4926;
+		ARG 0 property
+		ARG 1 trueModel
+		ARG 2 falseModel
+	METHOD method_25566 createValueMarkedVariantMap (Lnet/minecraft/class_2769;Ljava/lang/Comparable;Lnet/minecraft/class_2960;Lnet/minecraft/class_2960;)Lnet/minecraft/class_4926;
```
Comment (Runemoro) [inline mappings/net/minecraft/data/client/model/BlockStateModelGenerator.mapping]: What does "value marked" mean?

File: mappings/net/minecraft/data/client/model/BlockStateModelGenerator.mapping [hunk 5]
```diff
@@ -1,12 +1,422 @@
...
+		ARG 2 textureGetter
+	METHOD method_25560 registerTopSoil (Lnet/minecraft/class_2248;Lnet/minecraft/class_2960;Lnet/minecraft/class_4935;)V
+		ARG 1 topSoil
+		ARG 2 modelId
+		ARG 3 snowyVariant
+	METHOD method_25561 registerPiston (Lnet/minecraft/class_2248;Lnet/minecraft/class_2960;Lnet/minecraft/class_4944;)V
+		ARG 1 piston
+		ARG 2 extendedModelId
+		ARG 3 texture
+	METHOD method_25565 createBooleanPropertyBasedModelMap (Lnet/minecraft/class_2746;Lnet/minecraft/class_2960;Lnet/minecraft/class_2960;)Lnet/minecraft/class_4926;
```
Comment (Runemoro) [inline mappings/net/minecraft/data/client/model/BlockStateModelGenerator.mapping]: `createBooleanModelMap`

File: mappings/net/minecraft/data/client/model/BlockStateModelGenerator.mapping [hunk 6]
```diff
@@ -1,12 +1,422 @@
...
+		ARG 0 block
+		ARG 1 firstModelId
+		ARG 2 secondModelId
+	METHOD method_25646 createStairsBlockState (Lnet/minecraft/class_2248;Lnet/minecraft/class_2960;Lnet/minecraft/class_2960;Lnet/minecraft/class_2960;)Lnet/minecraft/class_4917;
+		ARG 0 stairsBlock
+		ARG 1 innerModelId
+		ARG 2 regularModelId
+		ARG 3 outerModelId
+	METHOD method_25649 createAxisRotatedVariantMap ()Lnet/minecraft/class_4926;
+	METHOD method_25650 registerCubeAllDecorative (Lnet/minecraft/class_2248;)Lnet/minecraft/class_4910$class_4912;
```
Comment (Runemoro) [inline mappings/net/minecraft/data/client/model/BlockStateModelGenerator.mapping]: "cube all decorative"? What does "all" mean here? Should it just be `registerDecorativeCube`?

File: mappings/net/minecraft/data/client/model/BlockStateModelGenerator.mapping [hunk 7]
```diff
@@ -1,12 +1,422 @@
...
+			ARG 2 modelId
+			ARG 3 block
+		METHOD method_25713 includeWithItem (Lnet/minecraft/class_4942;[Lnet/minecraft/class_2248;)Lnet/minecraft/class_4910$class_4911;
+			ARG 1 model
+			ARG 2 blocks
+		METHOD method_25714 includeWithItem ([Lnet/minecraft/class_2248;)Lnet/minecraft/class_4910$class_4911;
+			ARG 1 blocks
+		METHOD method_25715 includeWithoutItem ([Lnet/minecraft/class_2248;)Lnet/minecraft/class_4910$class_4911;
+			ARG 1 blocks
+	CLASS class_4912 DecorativeDerivative
```
Comment (Runemoro) [inline mappings/net/minecraft/data/client/model/BlockStateModelGenerator.mapping]: This seems to include things like buttons, signs, fences, pressure plates... Those aren't really decorative. Also, why is this not `DecorativeBlockDerivative`, like `BuiltinBlockDerivative`?

File: mappings/net/minecraft/data/client/model/BlockStateModelGenerator.mapping (hunk 4 above)
Comment (liach) [reply mappings/net/minecraft/data/client/model/BlockStateModelGenerator.mapping]: The value serves as a fence; a value in the property higher than to equal to this mark (I should probably rename to fence) value would give the variant map (a variant map is a subset of block state properties to a subset of model variant parameters (model, rotations, etc), such as powered -> model, facing -> x/y, etc) a model that is `higherModel`, while the other values, less than the mark, would be associated their model be this `lowerModel`.

File: mappings/net/minecraft/data/client/model/BlockStateModelGenerator.mapping (hunk 7 above)
Comment (liach) [reply mappings/net/minecraft/data/client/model/BlockStateModelGenerator.mapping]: This one generates block states and models from a base block's texture. Dunno what a better name could be. If this is not `Decorative`, what name do you suggest?

File: mappings/net/minecraft/data/client/model/BlockStateModelGenerator.mapping (hunk 6 above)
Comment (liach) [reply mappings/net/minecraft/data/client/model/BlockStateModelGenerator.mapping]: it registers a block with cube_all model and returns a decorative derivative which you can use to register other things like fences, buttons, etc.

Comment (liach) [reply mappings/net/minecraft/data/client/model/BlockStateModelGenerator.mapping]: Done.

File: mappings/net/minecraft/data/client/model/BlockStateModelGenerator.mapping (hunk 7 above)
Comment (liach) [reply mappings/net/minecraft/data/client/model/BlockStateModelGenerator.mapping]: Done. now called block texture pool

**Review by Runemoro — APPROVED** [review]

**Review by modmuss50 — APPROVED** [review]
